rholang/src/main/scala/coop/rchain/rholang/interpreter/SystemProcesses.scala

```diff
@@ -221,16 +221,15 @@ object SystemProcesses {
         phloRate  <- shortLeashParams.phloRate.get
         userId    <- shortLeashParams.userId.get
         timestamp <- shortLeashParams.timestamp.get
-      } yield {
-        space
-          .produce(
-            ack,
-            ListParWithRandom(Seq(codeHash, phloRate, userId, timestamp), rand),
-            false
-          )(MATCH_UNLIMITED_PHLOS)
-          .map(unpackOption(_))
-          .foldResult(dispatcher)
-      }
+        _ <- space
+              .produce(
+                ack,
+                ListParWithRandom(Seq(codeHash, phloRate, userId, timestamp), rand),
+                false
+              )(MATCH_UNLIMITED_PHLOS)
+              .map(unpackOption(_))
+              .foldResult(dispatcher)
+      } yield ()
```

Comment:
Yes. You can see that `getDeployParams` used to have `space.produce` being `yield`ed, but that was wrong because `space.produce` returned a `Task[Unit]`. This is why Kent has refactored to pull it into the `for`-comprehension. This is exactly why we should turn on `-Ywarn-value-discard`, then Scala won't silently throw away values and this kind of thing will get caught by the compiler. We should probably make a ticket to turn that on and fix all the problems it finds.

Reply:
Created https://rchain.atlassian.net/browse/RHOL-1001

Reply:
That's because space.produce returned an Option when I wrote it.

Reply:
@birchmd I had created https://rchain.atlassian.net/browse/RHOL-998 but I like your ticket description better